priv/samples/basics/button.livemd

```diff
@@ -0,0 +1,96 @@
+# Button Input using Circuits GPIO
+
+## Introduction
+
+In this exercise, we will use the [circuits_gpio] library to control 
+a GPIO as an input and output, turning an LED on and off with a button press.
+
+## Try it out
+
+Lets set a variable for the LED and button so that we dont have to choose the GPIO pin every time.
+
+First, refer to the GPIO ports for your nerves device.
+
+![](https://raw.githubusercontent.com/fhunleth/nerves_livebook/main/assets/gpio/rpi_pinout.png)
+
+```elixir
+led_pin = 20
+button_pin = 17
+```
+
+Alright! Now that we've selected our pins, make sure that we connect
+the to our `led_pin` and `ground` through a 220Ω resistor. The long leg
+can be connected to our `led_pin` and the short pin can connect from 
+the shorter leg to a 220Ω resistor, then to our ground.
+
+Connect one of the legs of the button to a 10kΩ resistor to 5v, and the diagonal leg to your `button_pin`.
+
+![Frizting Diagram](https://raw.githubusercontent.com//nerves_livebook/main/assets/gpio/button_and_led_rpi4_rpi0.svg)
+
+The following code will configure the `led_pin` as an `:output` and the `button_pin` as an `input`. 
+
+```elixir
+{:ok, led_output} = Circuits.GPIO.open(led_pin, :output)
+{:ok, button_input} = Circuits.GPIO.open(button_pin, :input)
+```
+
+Whenever you run the following code, it will read wheather your button is pressed or not, and set the `led_output` to match. 
+
+```elixir
+button_state = Circuits.GPIO.read(button_input)
+Circuits.GPIO.write(led_output, button_state)
+```
+
+This adeptation of the [hello_gpio] example will toggle the `output_gpio` to the `input_gpio` state.
+
+Be sure the `start` functions `GPIO.open` match your setup above.
+
+```elixir
+defmodule HelloGpio do
```

Comment:
```suggestion
defmodule HelloGPIO do
```

I realize that this may not be consistent everywhere in Nerves. The convention changed a couple years back to capitalize acronyms and I've been trying to do that going forward.

Reply:
Completely makes sense, updated in a second commit.